MoltenVK/MoltenVK/GPUObjects/MVKDeviceMemory.mm

```diff
@@ -105,30 +105,45 @@
 VkResult MVKDeviceMemory::addBuffer(MVKBuffer* mvkBuff) {
 	lock_guard<mutex> lock(_rezLock);
 
+	// If a dedicated alloc, ensure this buffer is the one and only buffer
+	// I am dedicated to.
+	if (_isDedicated && (_buffers.empty() || _buffers[0] != mvkBuff) )
+		return VK_ERROR_OUT_OF_DEVICE_MEMORY;
```

Comment:
When returning errors, MoltenVK convention is to log error text where appropriate using `mvkNotifyErrorWithText()` (eg. see 3 lines below). Is there a reason why you are not logging the error text here?